src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -248,14 +242,47 @@ void CRouterRoutino::buildDatabaseList()
                 continue;
             }
 
+            // qDebug() << "buildDatabase" << prefix;
+
 #ifdef Q_OS_WIN
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toLocal8Bit(), prefix.toLocal8Bit());
 #else
             Routino_Database * data = Routino_LoadDatabase(dir.absolutePath().toUtf8(), prefix.toUtf8());
 #endif
+            QVariantMap dmap;
+
             if(data)
             {
-                comboDatabase->addItem(prefix.replace("_", " "), quint64(data));
+                /* determine the profile to use */
+                dmap["db"] = QVariant ((qulonglong)data);
+
+                IAppSetup *setup = IAppSetup::getPlatformInstance();
+
+                /* check possible profiles.xml locations and keep the first active one. */
+                dmap["prefixpath"] = "";
+                QStringList profilespaths = {
+                    dir.filePath(prefix+"-profiles.xml"),
+                    dir.filePath("profiles.xml"),
+                    setup->routinoPath("profiles.xml").toUtf8()
```

Comment:
This path won't change over all iterations.

Reply:
No it doesnt but  I think it is cleaner to keep it in the check list for each database rather than handling an if clause.